runtime/src/iree-amd-aie/driver/xrt/pipeline_layout.h

```diff
@@ -14,6 +14,35 @@
 extern "C" {
 #endif  // __cplusplus
 
+// The max number of bindings per descriptor set allowed in the XRT HAL
+// implementation.
+#define IREE_HAL_XRT_MAX_DESCRIPTOR_SET_BINDING_COUNT 16
+
+// The max number of descriptor sets allowed in the XRT HAL implementation.
+//
+// This depends on the general descriptor set planning in IREE and should adjust
+// with it.
+#define IREE_HAL_XRT_MAX_DESCRIPTOR_SET_COUNT 4
+
+// Note that IREE HAL uses a descriptor binding model for expressing resources
+// to the kernels--each descriptor specifies the resource information, together
+// with a (set, binding) number indicating which "slots" it's bound to.
+//
+// In XRT, however, we don't have a direct correspondance of such mechanism.
+// Resources are expressed as kernel arguments. Therefore to implement IREE
+// HAL descriptor set and pipepline layout in XRT, we order and flatten all
+// sets and bindings and map to them to a linear array of kernel arguments.
+//
+// For example, given a pipeline layout with two sets and two bindings each:
+//   (set #, binding #) | kernel argument #
+//   :----------------: | :---------------:
+//   (0, 0)             | 2
```

Comment:
as mentioned on line 43, they are to do with the LX6 ASM instructions, I should put that above rather than below though, will change that in the next iteration.